src/xpk/commands/cluster.py

```diff
@@ -696,6 +697,146 @@ def cluster_create_ray_cluster(args) -> None:
+    )
+    xpk_print(f"[XPK] Task: 'Clone deployment ' in progress, Target directory:{coredns_repo_full_path}.")
+    return_code = run_command_with_updates(
+      command_git_clone, 'Clone deployment', args
+    )
+    if return_code != 0:
+      xpk_print(f'[XPK] Clone deployment error {return_code}')
+      xpk_exit(return_code)
+
+
+  cluster_name = os.environ.get('CLUSTER_NAME')
+  region = os.environ.get('REGION')
+  project = os.environ.get('PROJECT')          
+
+  if not all([cluster_name, region, project]):
+    missing_vars = []
+    if not cluster_name: missing_vars.append('CLUSTER_NAME')
+    if not region: missing_vars.append('REGION')
+    if not project: missing_vars.append('PROJECT')
+    xpk_print(f"[XPK] Error: Missing required environment variable:{', '.join(missing_vars)}. Please ensure these variables are set.")
+    xpk_exit(1)
+
+  command_get_credentials = (
+    f'gcloud container clusters get-credentials {cluster_name} --dns-endpoint '
+    f'--region={region} --project={project} && kubectl config view '
+    f'&& kubectl config set-context --current --namespace=default'
+  )
+  xpk_print(f"[XPK] Task: 'Get cluster credentials' in progress.")
+  return_code = run_command_with_updates(
+    command_get_credentials, 'Get cluster credentials', args
+  )
+  if return_code != 0:
+    xpk_print(f'[XPK] Failed to get cluster credentials {return_code}')
+    xpk_exit(return_code)
+
+  if not os.path.isdir(coredns_k8s_path):
+    xpk_print(f"[XPK] Error：CoreDNS Kubernetes path '{coredns_k8s_path}' does not exist. Has git clone been successful?")
+    xpk_exit(1)
+
+  # Remember the current directory so that you can restore it later
+  original_cwd = os.getcwd()
+  try:
+    # Change the current working directory to the path of the CoreDNS deployment
+    os.chdir(coredns_k8s_path)
+
+    command_deploy_coredns = (
+        f'./deploy.sh | kubectl apply -f -'
+    )
+    xpk_print(f"[XPK] Task: 'Deploy CoreDNS' in progress, Located at '{coredns_k8s_path}'")
+    
+    return_code = run_command_with_updates(
+        command_deploy_coredns, 'Deploy CoreDNS', args
+    )
+    if return_code != 0:
+      xpk_print(f'[XPK] Deploy CoreDNS error {return_code}')
+      xpk_exit(return_code)
+
+  finally:
+      # Whether it succeeds or fails, always restore to the original directory
+      os.chdir(original_cwd)
+
+  # Scale down kube-dns-autoscaler
+  command_autoscaler_scale_down = (
+    'kubectl scale deployment kube-dns-autoscaler --replicas=0 --namespace=kube-system'
+  )
+  xpk_print(f"[XPK] Task: 'Scaling down kube-dns-autoscaler' in progress")
+  return_code = run_command_with_updates(
+    command_autoscaler_scale_down, 'Scale down kube-dns-autoscaler', args
+  )
+  if return_code != 0:
+    xpk_print(f'[XPK] Scale down kube-dns-autoscaler error {return_code}')
+    xpk_exit(return_code)
+  xpk_print("\n[XPK] kube-dns-autoscaler has been scaled down.")
```

Comment:
I'd just add a comment here to note that the scale down command has started, but that scale down itself may take some time, and same with the one below.